packages/mco/hooks/protected-applications.ts

```diff
@@ -0,0 +1,190 @@
+import * as React from 'react';
+import { getNamespace } from '@odf/shared/selectors';
+import { useK8sWatchResources } from '@openshift-console/dynamic-plugin-sdk';
+import { RAMEN_PROTECTED_APPS_NAMESPACE } from '../constants';
+import { ACMPlacementModel } from '../models';
+import {
+  ACMSubscriptionKind,
+  ArgoApplicationSetKind,
+  DRPlacementControlKind,
+} from '../types';
+import { findPlacementNameFromAppSet, isPlacementModel } from '../utils';
+import {
+  getApplicationSetResourceObj,
+  getSubscriptionResourceObj,
+  getDRPlacementControlResourceObj,
+} from './mco-resources';
+
+const getPlacementUniqueName = (name, kind, namespace) =>
+  `${name}-${kind}-${namespace}`;
+
+const getResources = () => ({
+  argoApplicationSets: getApplicationSetResourceObj(),
+  subscriptions: getSubscriptionResourceObj(),
+  drPlacementControls: getDRPlacementControlResourceObj(),
+});
+
+export const useApplicationsWatch: UseApplicationsWatch = () => {
+  const response = useK8sWatchResources<WatchResourceType>(getResources());
+
+  const {
+    data: argoAppSets,
+    loaded: argoAppSetsLoaded,
+    loadError: argoAppSetsLoadError,
+  } = response?.argoApplicationSets;
+
+  const {
+    data: subscriptions,
+    loaded: subscriptionsLoaded,
+    loadError: subscriptionsLoadError,
+  } = response?.subscriptions;
+
+  const {
+    data: drpcs,
+    loaded: drpcsLoaded,
+    loadError: drpcsLoadError,
+  } = response?.drPlacementControls;
+
+  const drpcLoadedWAError = drpcsLoaded && !drpcsLoadError;
+  const subsLoadedWAError = subscriptionsLoaded && !subscriptionsLoadError;
+  const argoAppSetsLoadedWAError = argoAppSetsLoaded && !argoAppSetsLoadError;
+
+  const [protectedPlacementMap, drPolicyToDiscoveredAppsCount] =
+    React.useMemo(() => {
+      const protectedPlacementMap: {
+        [key: string]: Set<DRPlacementControlKind>;
+      } = {};
```

Comment:
```suggestion
      const protectedPlacementMap: {
        [key: string]: DRPlacementControlKind;
      } = {};
```
I did a mistake while explaining, per placement unique name there can't be more than one DRPC